### DISCUSSION TITLE
User is lost when being sent to my app from external pages

### DISCUSSION THREAD

**Discussion by slavaboiko** [gladly-team/next-firebase-auth#575] [Q&A] [answered]
Example:

Somewhere on external site
```
<a href="http://my.web.site/some/page">Go to my app</a>
```

My page:

```
const MyComponent = () => {
  return (<><h1>Hello</h1></>);
}

export default withAuthUser({
  whenAuthed: AuthAction.RENDER,
  whenUnauthedBeforeInit: AuthAction.SHOW_LOADER,
  whenAuthedBeforeRedirect: AuthAction.SHOW_LOADER,
  whenUnauthedAfterInit: AuthAction.REDIRECT_TO_LOGIN,
  LoaderComponent: FullPageLoader,
})(MyComponent);
```

SSR:

```
export const getServerSideProps =  withAuthUserTokenSSR({
  whenAuthed: AuthAction.RENDER,
  authPageURL: "/login"
})(handler);
```

my handler:

```
...
console.log("user", context.AuthUser);
...
```

produces:

```
user {
  id: null,
  email: null,
  emailVerified: false,
  phoneNumber: null,
  displayName: null,
  photoURL: null,
  claims: {},
  getIdToken: [Function (anonymous)],
  clientInitialized: false,
  firebaseUser: null,
  signOut: [Function (anonymous)],
  serialize: [Function: serialize]
}
```

And if I refresh the page - everything loads correctly. No idea why user is lost when coming from external pages. Tried searching PRs and Discussions - can't find anything...

**Comment by kmjennison** [comment, marked as answer]
This behavior is likely determined by your cookie's `SameSite` setting:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#values

I assume you're using `Strict`? Try `Lax` and see if it changes the behavior.

**Reply by slavaboiko** [reply]
> This behavior is likely determined by your cookie's `SameSite` setting: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#values
> 
> I assume you're using `Strict`? Try `Lax` and see if it changes the behavior.

yes! omg, it took me so long figuring this out and it was just a little config haha, thanks a lot!
